curriculum/challenges/english/25-front-end-development/workshop-recipe-tracker/66fbcf750a62784cf11f5635.md

```diff
@@ -0,0 +1,117 @@
+---
+id: 66fbcf750a62784cf11f5635
+title: Step 11
+challengeType: 0
+dashedName: step-11
+---
+
+# --description--
+
+To finally fill up each item of the `recipes` array with values for the `averageRating`, `totalIngredients`, and `difficultyLevel` properties, you can use a `for...of` loop .
+
+Here's the basic syntax of a `for...of` loop:
+
+```js
+for (const variable of iterable) {
+  // Code to execute for each element in the iterable
+}
+```
+
+And here's an example:
+
+```js
+const fruits = ['apple', 'banana', 'orange'];
+
+for (let fruit of fruits) {
+  console.log(fruit); // apple \n banana \n orange
+}
+```
+
+Create a `for...of` loop with `recipe` as the variable and `recipes` as the iterable. Leave it empty for now.
```

Comment:
Looking at the sheet, I didn't realize loops haven't been taught yet. 
So we can't use a for of loop here

Update both steps 11 and 12 to have them update just the first few properties manually

```md
recipe1.averageRating = getAverageRating(recipe1.ratings);
recipe1.totalIngredients = getTotalIngredients(recipe1.ingredients);
recipe1.difficultyLevel = getDifficultyLevel(recipe1.cookingTime);

```
 then in the final step have supply the rest for them and mention that they will learn how to work with loops which will make it easier to update each of these

Reply:
The issue is that the current examples doesn't actual teach them.
The basic syntax shows them how to get through the step but they won't understand what is going on because this will be their first introduction into loops.

So if we are going to keep this here, this needs to be updated to teach loops and how the for of loop actual works like what does `for (const variable of iterable)` actually mean? 

So a more robust example that teaches them and mentions that they will learn more about loops in the next module is fine